pysparkle/pysparkle/pyspark_utils.py

```diff
@@ -0,0 +1,92 @@
+# Spark common utilities
+import logging
+from typing import Any
+
+from pyspark.sql import DataFrame, SparkSession
+
+from pysparkle.utils import substitute_env_vars
+
+logger = logging.getLogger(__name__)
+
+
+def create_spark_session(app_name: str, spark_config: dict[str, Any] = None) -> SparkSession:
+    """Creates a SparkSession with the specified application name or retrieves one that already exists.
+
+    The SparkSession is configured to include Hadoop Azure and Delta Lake packages. If a SparkSession is already active,
+    it will be returned instead of creating a new one.
+
+    Args:
+        app_name: Name of the Spark application.
+        spark_config: A dictionary with additional Spark configuration options to set.
+
+    Returns:
+        Spark session with Delta Lake as the catalog implementation.
+
+    """
+    spark = SparkSession.getActiveSession()
+    if not spark:
+        logger.info("Creating Spark session...")
+
+        config = {
+            "spark.jars.packages": "org.apache.hadoop:hadoop-azure:3.3.4,io.delta:delta-core_2.12:2.4.0",
+            "spark.sql.extensions": "io.delta.sql.DeltaSparkSessionExtension",
+            "spark.sql.catalog.spark_catalog": "org.apache.spark.sql.delta.catalog.DeltaCatalog",
```

Comment:
Might be good to set these in a constants file.

Reply:
Personally I like having specific config like this defined somewhere a more explicit, e.g. a `SPARK_DEFAULT_CONFIG` constant, which could then be the default argument for `spark_config`. I can live with it though if you prefer it like this.

Reply:
I'll update it slightly :)

Reply:
On second thought I prefer that it stay as is. I could extract the constant for the defaults, but I also want the function to enable adding custom configs, which update the initial dictionary. Since a dictionary is mutable, I'd have to do a shallow copy, which makes things too complicated if we have just one place that uses this configuration. If we need this configuration to be accessed by more functions, I'd extract it then. For now I'd say is a premature optimisation :)